registry/api/v2/errors.go

```diff
@@ -15,6 +15,7 @@ var (
 		Message: "The operation is unsupported.",
 		Description: `The operation was unsupported due to a missing
 		implementation or invalid set of parameters.`,
+		HTTPStatusCode: http.StatusMethodNotAllowed,
```

Comment:
Are we okay with saying that whenever an `ErrorCodeUnsupported` (an operation is not supported), the API will behave as if the method is not supported?

I think this is a fine constraint but want to make sure we agree this works in all cases, which I think it does.